tools/unicode_properties_parse/format_width_estimate_intervals.cpp

```diff
@@ -0,0 +1,204 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+// The following code generates data for `_Width_estimate_intervals` in <format>.
+
+#include <charconv>
+#include <format>
+#include <fstream>
+#include <iostream>
+#include <regex>
+#include <vector>
+
+void verify_impl(bool test, int line, const char* msg) {
+    if (!test) {
+        std::cerr << "Error at line " << line << ": " << msg << std::endl;
+        exit(EXIT_FAILURE);
+    }
+}
+#define verify(expr, msg) verify_impl((expr), __LINE__, (msg))
+constexpr const char* impl_assertion_failed = "impl assertion failed";
+
+struct range_u {
+    uint32_t from, to;
+    range_u(uint32_t f, uint32_t t) : from(f), to(t) {}
+    range_u(uint32_t v) : from(v), to(v) {}
+};
+
+enum class width_u : bool { is_1 = false, is_2 = true };
+
+class table_u {
+    // A valid Unicode code point won't exceed `max_u`.
+    static constexpr uint32_t max_u = 0x10'ffff;
+    std::vector<width_u> table;
+
+public:
+    table_u() : table(max_u + 1, width_u::is_1) {}
+    void fill_range(const range_u rng, const width_u width) {

```

Comment:
Style: We conventionally place newlines between different function definitions, except when they are small and highly repetitive. This makes it easier to see where definitions begin and end.